examples/basic_operations/add_responsive_search_ad.py

```diff
@@ -12,56 +12,270 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""This example adds a responsive search ad to a given ad group.
-
-To get ad groups, run get_ad_groups.py.
 """
+This example shows how to create a complete Responsive Search ad.
+
+Includes creation of: budget, campaign, ad group, ad group ad,
+keywords, geo targeting, and image extensions.
 
+More details on Responsive Search ads can be found here:
+https://support.google.com/google-ads/answer/7684791
+"""
 
 import argparse
 import sys
-from uuid import uuid4
+import uuid
+import requests
 
 from google.ads.googleads.client import GoogleAdsClient
 from google.ads.googleads.errors import GoogleAdsException
 
 
-def main(client, customer_id, ad_group_id):
-    ad_group_ad_service = client.get_service("AdGroupAdService")
+# Keywords from user.
+_KEYWORD_TEXT_EXACT_1 = "example of exact match"
```

Comment:
The use of an initial underbar is very confusing since variable that begin this way are handled in a specific way by the Python import system. Capital letters are sufficient to indicate global variables.